django/template/context.py

```diff
@@ -269,7 +269,10 @@ def make_context(context, request=None, **kwargs):
     Create a suitable Context from a plain dict and optionally an HttpRequest.
     """
     if request is None:
-        context = Context(context, **kwargs)
+        if isinstance(context, RequestContext):
+            raise TypeError('RequestContext object is not a valid parameter')
```

Comment:
Something like this may be an improvement:
```
if not isinstance(context, dict):
   raise TypeError('context must be a dict rather than %r.' % context.__class__)
```
At least the error message can be improved to explain the expected type.